WalletWasabi/Tor/Socks5/Pool/TorHttpPool.cs

```diff
@@ -30,15 +30,21 @@ public class TorHttpPool : IDisposable
 		/// <remarks>This parameter affects maximum parallelization for given URI host.</remarks>
 		public const int MaxConnectionsPerHost = 10;
 
+		public delegate Task<HttpResponseMessage> SendCoreDelegateAsync(TorTcpConnection connection, HttpRequestMessage request, CancellationToken token = default);
+
+		/// <seealso cref="SendCoreAsync(TorTcpConnection, HttpRequestMessage, CancellationToken)"/>
+		private readonly SendCoreDelegateAsync _sendCoreDelegateAsync;
+
 		public TorHttpPool(EndPoint endpoint)
+			: this(new TorTcpConnectionFactory(endpoint))
 		{
-			TcpConnectionFactory = new(endpoint);
 		}
 
-		/// <summary>Use in tests.</summary>
-		internal TorHttpPool(TorTcpConnectionFactory tcpConnectionFactory)
+		/// <summary>Constructor that helps in tests.</summary>
+		internal TorHttpPool(TorTcpConnectionFactory tcpConnectionFactory, SendCoreDelegateAsync? sendCoreDelegate = null)
 		{
 			TcpConnectionFactory = tcpConnectionFactory;
+			_sendCoreDelegateAsync = sendCoreDelegate ?? SendCoreAsync;
```

Comment:
How about virtual SendCoreAsync() and having a superclass in tests where you can overwrite?

Reply:
What about this approach instead https://github.com/zkSNACKs/WalletWasabi/pull/5751/commits/bcac0ec12a86441fc10673de32f1825f93c69d1f?diff=split&w=1? It does:

* Removes delegate and makes `SendCoreAsync` `internal` (your suggestion)
* Uses moq to replace implementation of `SendCoreAsync` for the test purposes (can be verified with MSVS debugger that it goes to correct methods).

This is IMO more flexible. WDYT?

Reply:
Awesome!